zookeeper-api/src/main/java/org/apache/helix/zookeeper/util/HttpRoutingDataReader.java

```diff
@@ -41,13 +42,17 @@
 
 
 public class HttpRoutingDataReader {
-  private static final String MSDS_ENDPOINT =
+  private static final String SYSTEM_MSDS_ENDPOINT =
       System.getProperty(MetadataStoreRoutingConstants.MSDS_SERVER_ENDPOINT_KEY);
   private static final int HTTP_TIMEOUT_IN_MS = 5000;
 
   /** Double-checked locking requires that the following fields be volatile */
-  private static volatile Map<String, List<String>> _rawRoutingData;
-  private static volatile MetadataStoreRoutingData _metadataStoreRoutingData;
+  // The following map stands for (MSDS endpoint, Raw Routing Data)
+  private static volatile Map<String, Map<String, List<String>>> _rawRoutingDataMap =
```

Comment:
In fact, if my understanding is correct, we people use this one, for one product, there is only one SYSTEM_MSDS_ENDPOINT would be defined. Say storagesystem1, or storagesystem2. They can't co-exist. Put it another way, this map the first key field would alway have only one value, right?

I guess the intention is that to make this httproutingdatareader servering multiple namespaces (storagesystem1, storagesysgtem2). In reality, they would still serve only one.

Or am I missing something here?

Reply:
Serves one namespaces. However, this is done this way to allow HttpRoutingDataReader to support multiple regions.